lib/puppet/type/iis_site.rb

```diff
@@ -314,6 +315,18 @@ def insync?(is)
     end
   end
 
+  newproperty(:authenticationinfo) do
```

Comment:
While we have resolved the validation, we now have a problem with idempotency. As shown below, the first run creates the site, the second run notifies that the authenticationinfo has changed, which it has not.

manifest:

```
iis_site { 'test_website':
  ensure             => 'started',
  applicationpool    => 'DefaultAppPool',
  physicalpath       => 'C:\inetpub\wwwroot\test_website',
  bindings             => [
    {
      'bindinginformation'   => '*:80:',
      'protocol'             => 'http',
    },
  ],
  authenticationinfo => {
    'basic'     => true,
    'anonymous' => false,
    'windows'   => false,
  },
}
```

Output:

```
PS C:\src\puppetlabs\modules\puppetlabs-iis> bundle exec puppet apply --modulepath C:\src\puppetlabs\modules f
oo.pp
Notice: Compiled catalog for winpupp-tvrrh3n.localdomain in environment production in 1.86 seconds
Notice: /Stage[main]/Main/Iis_site[test_website]/ensure: defined 'ensure' as 'started'
Notice: Applied catalog in 17.11 seconds
PS C:\src\puppetlabs\modules\puppetlabs-iis> bundle exec puppet apply --modulepath C:\src\puppetlabs\modules f
oo.pp
Notice: Compiled catalog for winpupp-tvrrh3n.localdomain in environment production in 1.54 seconds
Notice: /Stage[main]/Main/Iis_site[test_website]/authenticationinfo: authenticationinfo changed {
  'basic' => true,
  'anonymous' => false,
  'windows' => false,
  'iisClientCertificateMapping' => false,
  'digest' => false,
  'clientCertificateMapping' => false
} to {
  'basic' => true,
  'anonymous' => false,
  'windows' => false
}
Notice: Applied catalog in 7.94 seconds
```

At point of self.instances, authenticationinfo gets populated with correct info, but when it's 'merged' with what is specified in the manifest. In the manifest we want to allow a user to specify a subset instead of having to declare all options each time.

Reply:
This is now fixed with a modification to self.prefetch that merges the resource and provider hashes. The information is the same at this point, so there is no data loss, we're just making the manifest hash the same as the instances hash